File: swift/Tests/AttomusOTPTests/TestSupport.swift
import Foundation

@testable import AttomusOTP

enum TestSupport {
    static func asciiData(_ value: String) -> Data {
        Data(value.utf8)
    }

    static func repeatedData(count: Int, seed: UInt8 = 0xAB) -> Data {
        Data((0..<count).map { UInt8(truncatingIfNeeded: Int(seed) + $0) })
    }
}

File: kotlin/src/test/kotlin/com/attomus/otp/TestSupport.kt
package com.attomus.otp

import java.nio.charset.StandardCharsets

internal object TestSupport {
    fun asciiData(value: String): ByteArray = value.toByteArray(StandardCharsets.US_ASCII)

    fun repeatedData(count: Int, seed: Int = 0xAB): ByteArray {
        return ByteArray(count) { index -> (seed + index).toByte() }
    }
}

